sdk/identity/Azure.Identity/src/MsalConfidentialClient.cs

```diff
@@ -75,7 +76,7 @@ protected override async ValueTask<IConfidentialClientApplication> CreateClientA
             if (_appTokenProviderCallback != null)
             {
                 confClientBuilder.WithAppTokenProvider(_appTokenProviderCallback)
-                    .WithAuthority(_authority.AbsoluteUri, TenantId, false)
+                    .WithAuthority("https://aka.ms/azsdk/net/identity/managedidentitycredential/troubleshoot", TenantId, false)
```

Comment:
Not sure about this. MSAL tries to parse the authority to determine if it's AAD or ADFS or B2C. It might fallback to AAD here, but it seems brittle.